$NetBSD: patch-bin_mvn,v 1.4 2017/06/20 11:50:28 jperkin Exp $

Darwin's java_home needs to be evaluated not set.
Follow pkgsrc path convention.

--- bin/mvn.orig	2014-08-11 20:59:27.000000000 +0000
+++ bin/mvn
@@ -83,7 +83,7 @@ case "`uname`" in
              #
              # Apple JDKs
              #
-             export JAVA_HOME=/usr/libexec/java_home
+             export JAVA_HOME=`/usr/libexec/java_home`
            fi
            ;;
 esac
@@ -94,6 +94,15 @@ if [ -z "$JAVA_HOME" ] ; then
   fi
 fi
 
+if [ "$JAVA_HOME" = "" ] ; then
+  JAVA_HOME=@PKG_JAVA_HOME@
+fi
+
+if [ "$M2_HOME" = "" ] ; then
+  # try to find maven
+  M2_HOME=@PREFIX@
+fi
+
 if [ -z "$M2_HOME" ] ; then
   ## resolve links - $0 may be a link to maven's home
   PRG="$0"
@@ -196,7 +205,9 @@ export MAVEN_CMD_LINE_ARGS
 
 exec "$JAVACMD" \
   $MAVEN_OPTS \
-  -classpath "${M2_HOME}"/boot/plexus-classworlds-*.jar \
-  "-Dclassworlds.conf=${M2_HOME}/bin/m2.conf" \
+  -classpath "${M2_HOME}"/lib/java/maven/boot/plexus-classworlds-*.jar \
+  "-Dclassworlds.conf=@PKG_SYSCONFDIR@/m2.conf" \
   "-Dmaven.home=${M2_HOME}"  \
-  ${CLASSWORLDS_LAUNCHER} "$@"
\ No newline at end of file
+  "-Dorg.apache.maven.global-settings=@PKG_SYSCONFDIR@/settings.xml" \
+  "-Dorg.apache.maven.global-plugin-registry=@PKG_SYSCONFDIR@/plugin-registry.xml" \
+  ${CLASSWORLDS_LAUNCHER} "$@"
